subsys/bluetooth/host/conn.c

```diff
@@ -2835,6 +2835,24 @@ int bt_conn_auth_cb_register(const struct bt_conn_auth_cb *cb)
 	return 0;
 }
 
+int bt_conn_auth_cb_overlay(struct bt_conn *conn, const struct bt_conn_auth_cb *cb)
+{
+	/* The cancel callback must always be provided if the app provides
+	 * interactive callbacks.
+	 */
+	if (!cb->cancel &&
+	    (cb->passkey_display || cb->passkey_entry || cb->passkey_confirm ||
+	     cb->pairing_confirm)) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_SMP) && conn->type == BT_CONN_TYPE_LE) {
```

Comment:
Any reason why we are not guarding the entire function with `CONFIG_BT_SMP`?

Reply:
Ok, changed. I initially wanted to define the function and just return error code if not supported, but I noticed that `bt_conn_auth_cb_register` is guarded as you suggested, so I aligned my implementation.